core/block.js

```diff
@@ -657,19 +657,28 @@ Blockly.Block.prototype.getHue = function() {
 
 /**
  * Change the colour of a block.
- * @param {number|string} colour HSV hue value, or #RRGGBB string.
+ * @param {number|string} colour HSV hue value (0 to 360), #RRGGBB string,
+ *     or a message reference string pointing to one of those two values.
  */
 Blockly.Block.prototype.setColour = function(colour) {
-  var hue = Number(colour);
+  var dereferenced = goog.isString(colour) ?
+          Blockly.utils.replaceMessageReferences(colour) : colour;
```

Comment:
Fix indent (+4 spaces from the line above; I think I've turned off checking on ternary expressions).